From bdd710ff939ad2452e85030d9a4e42b857707cbf Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 31 Mar 2021 23:48:51 +0300
Subject: [PATCH] [SW-15002] gaudi: security: dis iatu access once

Spare configuring the iATU access disable (including security) after it
was already done.
In the new LKD protocol, the iATU access is disabled on any command sent
via the PCIe, so it might be done multiple times.
Doing this multiple times is not a problem, but is also redundant, so do
this only once.

Change-Id: Iefe2ce37e6452ba049a8a1a80d4ada1eb26a3f8e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 957a49f..dab2317 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -411,12 +411,15 @@
 		regmap_writel(i, mmGIC_DISTRIBUTOR__6_GICD_CLRSPI_NSR);
 }
 
+static bool iatu_disabled;
+
 void preboot_disable_iatu_access(void)
 {
-	if (!preboot_security_is_enabled())
+	if (iatu_disabled || !preboot_security_is_enabled())
 		return;
 
+	preboot_security_dbi_lock();
 	preboot_conf_iatu_access(false);
 	preboot_ack_razwi_irq();
-	preboot_security_dbi_lock();
+	iatu_disabled = true;
 }
